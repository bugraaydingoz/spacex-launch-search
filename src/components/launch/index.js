export { Launch } from './Launch'
export { LaunchList } from './LaunchList'
export { LaunchLoading } from './LaunchLoading'
