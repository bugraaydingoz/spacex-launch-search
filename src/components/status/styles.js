import styled from 'styled-components'

export const StatusContainer = styled.div`
  display: flex;
  align-items: center;
  height: 24px;
  padding: 0 14px;
  font-size: 14px;
  font-weight: bold;
  border-radius: 999px;
  border: ${({ type }) => `2px solid var(--${type}-color)`};
  color: ${({ type }) => `var(--${type}-color)`};
`
