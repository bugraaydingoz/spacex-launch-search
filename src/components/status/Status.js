import React from 'react'

// Styles
import { StatusContainer } from './styles'

export function Status({ type }) {
  return <StatusContainer type={type}>{type.toUpperCase()}</StatusContainer>
}
