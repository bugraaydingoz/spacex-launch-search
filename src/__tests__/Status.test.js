import React from 'react'
import renderer from 'react-test-renderer'
import { Status } from '../components/status'

test('renders correctly', () => {
  const tree = renderer.create(<Status type="success" />).toJSON()
  expect(tree).toMatchSnapshot()
})
