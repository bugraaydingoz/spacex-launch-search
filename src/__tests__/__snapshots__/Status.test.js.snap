// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm YHoGC"
  type="success"
>
  SUCCESS
</div>
`;
